### PR TITLE
Use strings as value

### PR DESCRIPTION
**Pull request by osdnk**

### CODE REVIEW

File: android/src/main/java/com/swmansion/reanimated/nodes/TransformNode.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
     @Override
     public double getValue(NodesManager nodesManager) {
-      return nodesManager.getNodeValue(nodeID);
+      return (double) nodesManager.getNodeValue(nodeID);
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/reanimated/nodes/TransformNode.java]: transforms can have strings too (also add to one of example screens to see if `rotate` works ok)

File: android/src/main/java/com/swmansion/reanimated/nodes/SetNode.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import com.facebook.react.bridge.ReadableMap;
 import com.swmansion.reanimated.NodesManager;
 
-public class SetNode extends Node<Double> {
+public class SetNode extends Node {
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/reanimated/nodes/SetNode.java]: Consider removing <Double> from Node definition.

**Comment by ckknight** [comment]
This'll be fun to integrate into the type definitions 😄.

Bug found at https://github.com/kmagiera/react-native-reanimated/blob/72b70ac60970f570e915c67a4db584dff8e09695/android/src/main/java/com/swmansion/reanimated/nodes/OperatorNode.java#L148 - `res` is assumed to be a `Double`, when it could be a `String`, which seems useful for `defined` as an operator to have.

https://github.com/kmagiera/react-native-reanimated/blob/72b70ac60970f570e915c67a4db584dff8e09695/ios/Nodes/REAOperatorNode.m#L71
shares a similar issue

Also, it appears that `and` and `or` would work on strings in Android (checking against null and then `0.`) but possibly lead to an issue on iOS, as `[val doubleValue]` is used there.

That being said, I'm not an ObjC or Java programmer by trade, so take my caution with a grain of salt.

File: android/src/main/java/com/swmansion/reanimated/nodes/SetNode.java (hunk 2 above)
Comment (osdnk) [reply android/src/main/java/com/swmansion/reanimated/nodes/SetNode.java]: cosidered ¯\_(ツ)_/¯

**Comment by ckknight** [comment]
#61

File: ios/RNReanimated.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -98,6 +101,8 @@
 		440FEC0C209062F100EEC73A /* Nodes */ = {
 			isa = PBXGroup;
 			children = (
+				660A442A2119B83E006BFD5E /* REAConcatNode.h */,
```
Comment (kmagiera) [inline ios/RNReanimated.xcodeproj/project.pbxproj]: where is dis REAConcatNode.m

File: android/src/main/java/com/swmansion/reanimated/nodes/BezierNode.java [hunk 4]
```diff
@@ -83,7 +83,7 @@ public BezierNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
 
   @Override
   protected Double evaluate() {
-    Double in = mNodesManager.getNodeValue(mInputID);
+    Double in = (double) mNodesManager.getNodeValue(mInputID);
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/reanimated/nodes/BezierNode.java]: Capital letter Double

File: android/src/main/java/com/swmansion/reanimated/NodesManager.java [hunk 5]
```diff
@@ -22,6 +22,7 @@
 import com.swmansion.reanimated.nodes.BlockNode;
 import com.swmansion.reanimated.nodes.ClockNode;
 import com.swmansion.reanimated.nodes.ClockOpNode;
+import com.swmansion.reanimated.nodes.ConcatNode;
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/reanimated/NodesManager.java]: where is dis?

Comment (osdnk) [reply android/src/main/java/com/swmansion/reanimated/NodesManager.java]: it's hir, it's hir

File: android/src/main/java/com/swmansion/reanimated/nodes/OperatorNode.java [hunk 6]
```diff
@@ -137,15 +137,15 @@ public double evaluate(Node[] input) {
...
     @Override
-    public double evaluate(Node[] input) {
+    public Object evaluate(Node[] input) {
       return truthy(input[0].value()) ? 0. : 1.;
     }
   };
   private static final Operator DEFINED = new Operator() {
     @Override
-    public double evaluate(Node[] input) {
+    public Object evaluate(Node[] input) {
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/reanimated/nodes/OperatorNode.java]: can we special case DEFINED in `evaluete` and leave out all the changes

Comment (osdnk) [reply android/src/main/java/com/swmansion/reanimated/nodes/OperatorNode.java]: indeed i figured out there's no need for any special case as the result of `defined` is double

**Pull request merged**